stlab/concurrency/channel.hpp

```diff
@@ -691,7 +691,7 @@ struct shared_process : shared_process_receiver<R>,
     bool                     _process_close_queue = false;
     // REVISIT (sparent) : I'm not certain final needs to be under the mutex
     bool                     _process_final = false;
-    std::shared_ptr<std::function<void()>> _process_timeout_function;
+    std::atomic_size_t       _process_timeout_control{0};
```

Comment:
Sorry - my comment got eaten by github. Should have been "maybe make an `atomic<enum class>` type if the values are only 0-2.